Add joining queries index page (#8088)
opensearch-project/documentation-website: _query-dsl/joining/index.md

---
layout: default
title: Joining queries
has_children: true
nav_order: 55
---

# Joining queries

> OpenSearch is a distributed system where data is spread across multiple nodes. Thus, running an SQL-like JOIN operation in OpenSearch is resource intensive. As an alternative, OpenSearch provides the following queries that perform join operations and are optimized for scaling across multiple nodes:

Reviewer: ```suggestion
OpenSearch is a distributed system in which data is spread across multiple nodes. Thus, running a SQL-like JOIN operation in OpenSearch can be resource intensive. As an alternative, OpenSearch provides the following queries that perform join operations and are optimized for scaling across multiple nodes:
```

- `nested` queries: Act as wrappers for other queries to search [nested]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/nested/) fields. The nested field objects are searched as though they were indexed as separate documents.

> - `has_child` queries: Search for parent documents whose child documents matching the query.

Reviewer 2: ```suggestion
- `has_child` queries: Search for parent documents whose child documents match the query.
```

> - `has_parent` queries: Search for child documents whose parent documents matching the query.

Reviewer 2: ```suggestion
- `has_parent` queries: Search for child documents whose parent documents match the query.
```
